core/math/vector2.cpp

```diff
@@ -45,6 +45,24 @@ real_t Vector2::length_squared() const {
 	return x * x + y * y;
 }
 
+void Vector2::set_axis(int p_axis, real_t p_value) {
+	ERR_FAIL_INDEX(p_axis, 2);
```

Comment:
Presumably we won't want these safety checks in the BVH, if using a template, this can never occur and will only slow it down. So better to set these values directly.

Reply:
Actually these functions and a number of these I'm not sure should be in the cpp as they are harder to inline / optimize for the compiler afaik (depending on the compiler). I'm not sure why they are in the cpp - short math type functions that don't draw in dependencies are usually better in the header for this reason.

Reply:
Makes sense, I'll change it to inline what can be inlined and avoid unnecessary checks.

Reply:
Actually, this wasn't even used in the bvh tree, I had just added it to make `Vector3` and `Vector2` the same. So now I've removed it from `Vector2`.